### PR TITLE
Add keypress stuffs

### PR DESCRIPTION
**Pull request by joker-119**
Allows players to manage having multiple on-use abilities in a single Custom Role easier, without the need for the old `cmdbind` system.

While cmdbind stuff will still work perfectly the same, this is a hugely breaking change to the ActiveAbility class, that will require active code changes in most plugins using them.

This allows active abilities to be used via simple keypresses, regardless of the role you are playing and how many abilities that role has.

Keypress guide:
1x Alt - Use currently selected ability (if any)
2x Alt - Switch to next ability (if possible)
C + 2x Alt - Switch to previous ability (if possible)
C + 1x Alt - Display information on current selected ability, including name, description, duration, cooldown and current usability status (will display remaining cooldown if it's on cooldown still)

Video Demonstration:
https://discord.com/channels/656673194693885975/668651891944587264/1141582398178459659

This PR will be marked as DNM until EXILED has begun accepting breaking changed for 8.0.0

### CODE REVIEW

**Comment by xNexusACS** [comment]
imagine copying other frameworks idea & stuff but changing the code

**Comment by louis1706** [comment]
> imagine copying other frameworks idea & stuff but changing the code, cringe, literally the synapse keypress system but edited

Bruh you really thinks than joker / nao was looking at synapse for copying code even me wasn't knowing that.

**Comment by joker-119** [comment]
> imagine copying other frameworks idea & stuff but changing the code, cringe, literally the synapse keypress system but edited

Coincidence since Synapse hasn't even been a thought in my mind since like.. SL 11.1, lol

File: Exiled.CustomRoles/API/Features/ActiveAbility.cs [hunk 1]
```diff
@@ -116,6 +187,29 @@ public virtual bool CanUseAbility(Player player, out string response)
...
+                AllActiveAbilities[player].Add(this);
+            base.AbilityAdded(player);
+        }
+
+        /// <inheritdoc />
+        protected override void AbilityRemoved(Player player)
+        {
+            if (!AllActiveAbilities.ContainsKey(player))
+                return;
+            if (SelectedPlayers.Contains(player))
```
Comment (louis1706) [inline Exiled.CustomRoles/API/Features/ActiveAbility.cs:206]: This check is not needed

File: Exiled.CustomRoles/API/Features/KeypressActivator.cs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        }
+
+        private IEnumerator<float> ProcessAltKey(Player player)
+        {
+            yield return Timing.WaitForSeconds(0.25f);
+
+            if (!altTracker.TryGetValue(player, out int pressCount))
+                yield break;
+
+            Log.Warn($"{player.Nickname}: {pressCount} {(player.Role is FpcRole fpc ? fpc.MoveState : false)}");
```
Comment (louis1706) [inline Exiled.CustomRoles/API/Features/KeypressActivator.cs:90]: Debug instead ?

File: Exiled.CustomRoles/API/Features/KeypressActivator.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                        Log.Warn("Joker can't do math.");
+                        response = "Jokey did a fucky wucky wif his maths";
+                        return false;
+                    }
+
+                    if (abilities[index] == selected)
+                    {
+                        response = "No abilities to switch to.";
+                        return false;
+                    }
```
Comment (louis1706) [inline Exiled.CustomRoles/API/Features/KeypressActivator.cs:154]: Why not just checking before the number of ability ? If( .Count <= 1) return false;

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Review by louis1706 — APPROVED** [review]

**Pull request merged**